### PR TITLE
save happiness and don't devidi through 0

### PR DESCRIPTION
**Pull request by Raycoms**
Hotfixes a crash

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 1]
```diff
@@ -346,6 +347,15 @@ private void readFromNBT(@NotNull final NBTTagCompound compound)
...
             freePositions.add(block);
         }
+
+        if(compound.hasKey(TAG_HAPPINESS))
+        {
+            this.overallHappiness = compound.getDouble(TAG_HAPPINESS);
+        }
+        else
+        {
+            this.overallHappiness = 5;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: MAJOR Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 58 issues
* ![MAJOR][MAJOR] 15 major
* ![MINOR][MINOR] 41 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/f1c32e66665521be3771ab3f91e64b076eb6f039/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.218 lines, which is greater than 750 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/f1c32e66665521be3771ab3f91e64b076eb6f039/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 42 to the maximum authorized 20 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/f1c32e66665521be3771ab3f91e64b076eb6f039/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 81 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/f1c32e66665521be3771ab3f91e64b076eb6f039/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Refactor this class so it has no more than 20 fields, rather than the 35 it currently has. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L165](https://github.com/Minecolonies/minecolonies/blob/f1c32e66665521be3771ab3f91e64b076eb6f039/src/main/java/com/minecolonies/coremod/colony/Colony.java#L165): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L616](https://github.com/Minecolonies/minecolonies/blob/f1c32e66665521be3771ab3f91e64b076eb6f039/src/main/java/com/minecolonies/coremod/colony/Colony.java#L616): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1134](https://github.com/Minecolonies/minecolonies/blob/f1c32e66665521be3771ab3f91e64b076eb6f039/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1134): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1135](https://github.com/Minecolonies/minecolonies/blob/f1c32e66665521be3771ab3f91e64b076eb6f039/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1135): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1248](https://github.com/Minecolonies/minecolonies/blob/f1c32e66665521be3771ab3f91e64b076eb6f039/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1248): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1250](https://github.com/Minecolonies/minecolonies/blob/f1c32e66665521be3771ab3f91e64b076eb6f039/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1250): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Review by marvin-bitterlich — DISMISSED** [review]
lol good to go

**Comment by marvin-bitterlich** [comment]
:shipit:

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 56 issues
* ![MAJOR][MAJOR] 13 major
* ![MINOR][MINOR] 41 minor
* ![INFO][INFO] 2 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/5d08f0a51037e895cde9762f1f11de0e6b00cac2/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.219 lines, which is greater than 750 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/5d08f0a51037e895cde9762f1f11de0e6b00cac2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 42 to the maximum authorized 20 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/5d08f0a51037e895cde9762f1f11de0e6b00cac2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 81 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/5d08f0a51037e895cde9762f1f11de0e6b00cac2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Refactor this class so it has no more than 20 fields, rather than the 35 it currently has. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L621](https://github.com/Minecolonies/minecolonies/blob/5d08f0a51037e895cde9762f1f11de0e6b00cac2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L621): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1139](https://github.com/Minecolonies/minecolonies/blob/5d08f0a51037e895cde9762f1f11de0e6b00cac2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1139): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1140](https://github.com/Minecolonies/minecolonies/blob/5d08f0a51037e895cde9762f1f11de0e6b00cac2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1140): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1253](https://github.com/Minecolonies/minecolonies/blob/5d08f0a51037e895cde9762f1f11de0e6b00cac2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1253): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1255](https://github.com/Minecolonies/minecolonies/blob/5d08f0a51037e895cde9762f1f11de0e6b00cac2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1255): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1257](https://github.com/Minecolonies/minecolonies/blob/5d08f0a51037e895cde9762f1f11de0e6b00cac2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1257): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Review by marvin-bitterlich — DISMISSED** [review]

**Review by marvin-bitterlich — APPROVED** [review]

**Comment by marvin-bitterlich** [comment]
:shipit:

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 73 issues
* ![MAJOR][MAJOR] 27 major
* ![MINOR][MINOR] 41 minor
* ![INFO][INFO] 5 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/c6b35c42d5be47d8c8137972fdd81bdae7e3ede9/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.219 lines, which is greater than 750 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/c6b35c42d5be47d8c8137972fdd81bdae7e3ede9/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 42 to the maximum authorized 20 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/c6b35c42d5be47d8c8137972fdd81bdae7e3ede9/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 81 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/c6b35c42d5be47d8c8137972fdd81bdae7e3ede9/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Refactor this class so it has no more than 20 fields, rather than the 35 it currently has. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L621](https://github.com/Minecolonies/minecolonies/blob/c6b35c42d5be47d8c8137972fdd81bdae7e3ede9/src/main/java/com/minecolonies/coremod/colony/Colony.java#L621): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1139](https://github.com/Minecolonies/minecolonies/blob/c6b35c42d5be47d8c8137972fdd81bdae7e3ede9/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1139): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1140](https://github.com/Minecolonies/minecolonies/blob/c6b35c42d5be47d8c8137972fdd81bdae7e3ede9/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1140): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1253](https://github.com/Minecolonies/minecolonies/blob/c6b35c42d5be47d8c8137972fdd81bdae7e3ede9/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1253): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1255](https://github.com/Minecolonies/minecolonies/blob/c6b35c42d5be47d8c8137972fdd81bdae7e3ede9/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1255): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1257](https://github.com/Minecolonies/minecolonies/blob/c6b35c42d5be47d8c8137972fdd81bdae7e3ede9/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1257): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**